File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)

        starts = []
        for num in nums_set:
            if num-1 not in nums_set:
                starts.append(num)

        result = 0
        for start in starts:
            num = start
            while num in nums_set:
                num += 1
            
            if num-start > result:
                result = num-start


        
        return result
                



        
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums: number[], k: number): number[] {
        const freqs = new Map<number, number>();
        for (const num of nums) {
            if (!freqs.has(num)) {
                freqs.set(num, 1);
                continue;
            }
            freqs.set(num, freqs.get(num)+1)
        }

        const groups: number[][]= Array.from({ length: nums.length + 1 }, () => []);
        for (const [num, freq] of freqs.entries()) {
            groups[freq].push(num);
        }

        const result: number[] = [];
        for (let i = nums.length; i >= 0; i--) {
            for (const n of groups[i]) {
                result.push(n);
                if (result.length === k) {
                    return result;
                }
            }
        }
    }
}
